### PR TITLE
ci: enable spotbugs for operate

### LINKED ISSUE #17331: Run Spotbugs plugin on the Operate codebase and fix related issues
### Description

Came up during integration Operate into the maven multi-module project, see here https://github.com/camunda/zeebe/pull/17280

It seems spotbugs were not executed before, for Operate. This leads to build errors when enabled.

Spotbugs allows us to detect several kind of bugs (based on [configuration](https://github.com/camunda/zeebe/blob/a2b58bb9041b8127ce336ca8c921b5ccac7e87b8/build-tools/src/main/resources/spotbugs/spotbugs-include.xml)) The current configuration is currently limited to [CORRECTNESS](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness) and [MT_CORRECTNESS](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#multithreaded-correctness-mt-correctness)

One we for example detected was #17316 but there are more. 

### TODO

Run spotbugs on the Operate code base and fix the related issues.

### Additional Information

Please reach out if you have any questions related to the tool.

---------------------------------------------------------------------------------------------

### Acceptance Criteria

- [ ] Checked if the specific data needs to be provided for the QA process

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered
- [ ] Cross-team dependencies have been considered

### LINKED ISSUE #17652: Create workflow for Operate that runs Java code checks
### Description

Create a workflow that runs the various Java check plugins on the operate codebase. For now this will mean running spotbugs (with the spotbugs profile) and adding -skipChecks on the other workflows. This will follow the same pattern established by Zeebe.

Existing spotbugs issues will need to be fixed as part of this, see issue #17331

### Additional Information

---------------------------------------------------------------------------------------------

### Acceptance Criteria

- [ ] Checked if the specific data needs to be provided for the QA process

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered
- [ ] Cross-team dependencies have been considered

### PR DESCRIPTION
**Pull request by matt-whiteman**
## Description

Suppressed warnings for 1 spotbugs issues:
- Usage of poll() without checking return value in SoftHashMap.trimStrongReferencesIfNecessary()(we only use poll to trim a list)

Fixed spotbugs issues:
- Added setters for fields flagged by spotbugs in AbstractStep
- Removed redundant null check in ImportJob.processPossibleIndexChange()
- Added null check for latestPosition variable in ElasticsearchRecordsReader.readAndScheduleNextBatch()
- Added null check for latestPosition variable  in OpensearchRecordsReader.readAndScheduleNextBatch()
- Removed redundant autowiring of RichOpenSearchClient in OpensearchOperationReader
- Change lock field in TokenAuthentication from Integer to static Object (static because Integer pooling implicitly made this a shared lock to begin with)

Added a new workflow 'Java Checks' to run spotbugs and other checks (licenses, format, etc)

## Related issues

closes #17331
closes #17652

### CODE REVIEW

**Comment by matt-whiteman** [comment]
These were all the spotbugs issues found and fixed:

[ERROR] Low: Return value of java.util.Queue.poll() ignored in io.camunda.operate.util.SoftHashMap.trimStrongReferencesIfNecessary() [io.camunda.operate.util.SoftHashMap] At SoftHashMap.java:[line 147] RV_RETURN_VALUE_IGNORED

[ERROR] Medium: Unwritten field: io.camunda.operate.schema.migration.AbstractStep.content [io.camunda.operate.schema.migration.AbstractStep] At AbstractStep.java:[line 42] UWF_UNWRITTEN_FIELD
[ERROR] Medium: Unwritten field: io.camunda.operate.schema.migration.AbstractStep.description [io.camunda.operate.schema.migration.AbstractStep] At AbstractStep.java:[line 47] UWF_UNWRITTEN_FIELD
[ERROR] Medium: Unwritten field: io.camunda.operate.schema.migration.AbstractStep.indexName [io.camunda.operate.schema.migration.AbstractStep] At AbstractStep.java:[line 37] UWF_UNWRITTEN_FIELD
[ERROR] Medium: Unwritten field: io.camunda.operate.schema.migration.AbstractStep.version [io.camunda.operate.schema.migration.AbstractStep] At AbstractStep.java:[line 52] UWF_UNWRITTEN_FIELD

[ERROR] Medium: Nullcheck of newImportBatch at line 109 of value previously dereferenced in io.camunda.operate.zeebeimport.ImportJob.processPossibleIndexChange() [io.camunda.operate.zeebeimport.ImportJob, io.camunda.operate.zeebeimport.ImportJob] At ImportJob.java:[line 109]Redundant null check at ImportJob.java:[line 111] RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE
[ERROR] Medium: Possible null pointer dereference of latestPosition in io.camunda.operate.zeebeimport.elasticsearch.ElasticsearchRecordsReader.readAndScheduleNextBatch(boolean) [io.camunda.operate.zeebeimport.elasticsearch.ElasticsearchRecordsReader, io.camunda.operate.zeebeimport.elasticsearch.ElasticsearchRecordsReader] Dereferenced at ElasticsearchRecordsReader.java:[line 175]Known null at ElasticsearchRecordsReader.java:[line 170] NP_NULL_ON_SOME_PATH
[ERROR] Medium: Possible null pointer dereference of latestPosition in io.camunda.operate.zeebeimport.opensearch.OpensearchRecordsReader.readAndScheduleNextBatch(boolean) [io.camunda.operate.zeebeimport.opensearch.OpensearchRecordsReader, io.camunda.operate.zeebeimport.opensearch.OpensearchRecordsReader] Dereferenced at OpensearchRecordsReader.java:[line 174]Known null at OpensearchRecordsReader.java:[line 169] NP_NULL_ON_SOME_PATH

[ERROR] Low: Field OpensearchOperationReader.richOpenSearchClient masks field in superclass io.camunda.operate.webapp.opensearch.reader.OpensearchAbstractReader [io.camunda.operate.webapp.opensearch.reader.OpensearchOperationReader] In OpensearchOperationReader.java MF_CLASS_MASKS_FIELD
[ERROR] Medium: Synchronization on Integer in io.camunda.operate.webapp.security.sso.TokenAuthentication.getAuthorizations() [io.camunda.operate.webapp.security.sso.TokenAuthentication] At TokenAuthentication.java:[line 140] DL_SYNCHRONIZATION_ON_BOXED_PRIMITIVE

File: operate/webapp/src/main/java/io/camunda/operate/webapp/security/sso/TokenAuthentication.java [hunk 1]
```diff
@@ -44,8 +44,14 @@ public class TokenAuthentication extends AbstractAuthenticationToken {
...
   public static final String ROLES_KEY = "roles";
-  private static Logger logger = LoggerFactory.getLogger(TokenAuthentication.class);
-  @JsonIgnore private final Integer lock = 0;
+  private static final Logger LOGGER = LoggerFactory.getLogger(TokenAuthentication.class);
+
+  // Need to research further whether a lock shared between all instances is the desired behavior.
+  // This was originally an Integer member variable (non-static) but due to integer pooling, it
+  // was effectively a shared instance. This static object mirrors that behavior to keep consistent
+  // behavior until we research further.
+  @JsonIgnore private static final Object RESOURCE_PERMISSIONS_LOCK = new Object();
```
Comment (sdorokhova) [inline operate/webapp/src/main/java/io/camunda/operate/webapp/security/sso/TokenAuthentication.java:53]: I think se don't really need `@JsonIgnore` for the static field? Or should we leave it for the case when we fix the problem discussed in comment? 🤔

File: operate/pom.xml [hunk 2]
```diff
@@ -170,20 +170,13 @@
...
-        See https://github.com/camunda/zeebe/issues/17331
-      -->
       <id>spotbugs</id>
       <build>
         <plugins>
           <plugin>
             <groupId>com.github.spotbugs</groupId>
             <artifactId>spotbugs-maven-plugin</artifactId>
             <version>${plugin.version.spotbugs}</version>
-            <configuration>
```
Comment (sdorokhova) [inline operate/pom.xml:184]: I see that `spotbugs-maven-plugin` is also declared in parent POM  and Zeebe is using `spotbugs` plugin to trigger the check (e.g. [here](https://github.com/camunda/zeebe/blob/57c33486e3cf0fcda20b04ccadd806583ad258f0/.github/workflows/zeebe-ci.yml#L504)). You decided not to go that way and rather have it always running and explicitly disabling by choice?

**Review by sdorokhova — APPROVED** [review]
Left couple of comments. But in overall looks good to me!

**Pull request merged**